aws/data_source_aws_workspaces_directory.go

```
@@ -0,0 +1,146 @@
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"increase_volume_size": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"rebuild_workspace": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"restart_workspace": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"switch_running_mode": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"subnet_ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"workspace_security_group_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"iam_role_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"registration_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"directory_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"directory_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"customer_user_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"alias": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ip_group_ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"dns_ip_addresses": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func dataSourceAwsWorkspacesDirectoryRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).workspacesconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	rawOutput, state, err := waiter.DirectoryState(conn, d.Id())()
+	if err != nil {
+		return fmt.Errorf("error getting WorkSpaces Directory (%s): %s", d.Id(), err)
+	}
+	if state == workspaces.WorkspaceDirectoryStateDeregistered {
```

When the resource isn't found, we need to return an error (#13410), otherwise there can be undefined behaviour.